tests/test_validated_input.ts

```diff
@@ -0,0 +1,193 @@
+import ValidatedInput from '@/components/validated_input.vue';
+import { zip } from '@/utils';
+import { config, mount } from '@vue/test-utils';
+
+beforeAll(() => {
+    config.logModifiedComponents = false;
+});
+
+const IS_NUMBER = (value: string): [boolean, string] => {
+    const valid = value !== "" && !isNaN(Number(value));
+    return [valid, "Invalid number!"];
+};
+
+describe('ValidatedInput.vue', () => {
+    test('Validated input uses default string conversion when to_string_fn not present',
+         () => {
+        const wrapper = mount(ValidatedInput, {
+            propsData: {
+                value: 1,
+                validators: [IS_NUMBER],
+                from_string_fn: (val: string) => parseInt(val, 10),
+            }
+        });
+
+        wrapper.vm.$mount();
+        const input: HTMLInputElement = <HTMLInputElement> wrapper.find('#input').element;
+        expect(input.value).toBe("1");
+    });
+
+    test('Validated input uses default when from_string_fn not present', () => {
+        const component = {
+            template:  `<validated-input ref="validated_input_1" v-model="my_input"
+                                         :validators="[() => [true, 'placeholder']]"/>`,
+            components: {
+                'validated-input': ValidatedInput
+            },
+            data: () => {
+                return {
+                    my_input: "hello"
+                };
+            }
+        };
+
+        const wrapper = mount(component);
+        wrapper.vm.$mount();
+        const validated_input = wrapper.find({ref: 'validated_input_1'});
+        expect(validated_input.vm.$data.input_value).toEqual("hello");
```

Comment:
As per above, change to `expect(validated_input.vm.input_value).toEqual("hello");`